### ISSUE TITLE
Using hyperparameters with Ordinal config_space in active_search_space

### ISSUE THREAD

**Issue by sighellan** [awslabs/syne-tune#500]
**Describe the bug**  
I want to use an `Ordinal` hyperparameter as the `active_task` in the `search_options` fed to `BayesianOptimization`. It works for `Categorical`, but gives assertion errors for `Ordinal`.

**To reproduce**  
```python
import syne_tune.config_space as sp
from syne_tune.optimizer.baselines import BayesianOptimization

def get_config_space(ordinal):
    config_space = {
        'parA': sp.loguniform(1e-6, 1.0),
        'parB': sp.loguniform(1e-5, 10),
    }
    if ordinal:
        config_space['problem_maker'] = sp.ordinal([1, 2, 3], kind='equal')
    else:
        config_space['problem_maker'] = sp.choice([1, 2, 3])
    return config_space


search_options = {'transfer_learning_task_attr': 'problem_maker',
                          'transfer_learning_active_task': 2,
                         }
for ordinal in [False, True]:
    print('Ordinal: %s' %ordinal)
    conf_space = get_config_space(ordinal)
    print(conf_space)
    scheduler = BayesianOptimization(
                                conf_space,
                                max_t=1,
                                mode="min",
                                metric='metric_error',
                                random_seed=1,
                                search_options=search_options
    )
    print('Works')
```

**Output**
```python
Ordinal: False
{'parA': loguniform(1e-06, 1.0), 'parB': loguniform(1e-05, 10), 'problem_maker': choice([1, 2, 3])}
Works
Ordinal: True
{'parA': loguniform(1e-06, 1.0), 'parB': loguniform(1e-05, 10), 'problem_maker': ordinal([1, 2, 3], kind='equal')}

---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
Cell In [1], line 23
     21 conf_space = get_config_space(ordinal)
     22 print(conf_space)
---> 23 scheduler = BayesianOptimization(
     24                             conf_space,
     25                             max_t=1,
     26                             mode="min",
     27                             metric='metric_error',
     28                             random_seed=1,
     29                             search_options=search_options
     30 )
     31 print('Works')

File ~/miniconda3/envs/syne-tune/lib/python3.8/site-packages/syne_tune/optimizer/baselines.py:62, in BayesianOptimization.__init__(self, config_space, metric, **kwargs)
     61 def __init__(self, config_space: dict, metric: str, **kwargs):
---> 62     super(BayesianOptimization, self).__init__(
     63         config_space=config_space,
     64         metric=metric,
     65         searcher="bayesopt",
     66         **kwargs,
     67     )

File ~/miniconda3/envs/syne-tune/lib/python3.8/site-packages/syne_tune/optimizer/schedulers/fifo.py:238, in FIFOScheduler.__init__(self, config_space, **kwargs)
    236     if "scheduler" not in search_options:
    237         search_options["scheduler"] = "fifo"
--> 238     self.searcher: BaseSearcher = searcher_factory(searcher, **search_options)
    239 else:
    240     assert isinstance(searcher, BaseSearcher)

File ~/miniconda3/envs/syne-tune/lib/python3.8/site-packages/syne_tune/optimizer/schedulers/searchers/searcher_factory.py:156, in searcher_factory(searcher_name, **kwargs)
    151     assert scheduler is not None, "Scheduler must set search_options['scheduler']"
    152     assert scheduler in supported_schedulers, (
    153         f"Searcher '{searcher_name}' only works with schedulers "
    154         + f"{supported_schedulers} (not with '{scheduler}')"
    155     )
--> 156 searcher = searcher_cls(**kwargs)
    157 return searcher

File ~/miniconda3/envs/syne-tune/lib/python3.8/site-packages/syne_tune/optimizer/schedulers/searchers/gp_fifo_searcher.py:603, in GPFIFOSearcher.__init__(self, config_space, metric, clone_from_state, **kwargs)
    601     kwargs["config_space"] = config_space
    602     kwargs["metric"] = metric
--> 603     kwargs_int = self._create_kwargs_int(kwargs)
    604 else:
    605     # Internal constructor, bypassing the factory
    606     # Note: Members which are part of the mutable state, will be
    607     # overwritten in `_restore_from_state`
    608     super().__init__(config_space, metric=metric)

File ~/miniconda3/envs/syne-tune/lib/python3.8/site-packages/syne_tune/optimizer/schedulers/searchers/gp_fifo_searcher.py:616, in GPFIFOSearcher._create_kwargs_int(self, kwargs)
    612 def _create_kwargs_int(self, kwargs):
    613     _kwargs = check_and_merge_defaults(
    614         kwargs, *gp_fifo_searcher_defaults(), dict_name="search_options"
    615     )
--> 616     kwargs_int = gp_fifo_searcher_factory(**_kwargs)
    617     # Extra arguments not parsed in factory
    618     self._copy_kwargs_to_kwargs_int(kwargs_int, kwargs)

File ~/miniconda3/envs/syne-tune/lib/python3.8/site-packages/syne_tune/optimizer/schedulers/searchers/gp_searcher_factory.py:487, in gp_fifo_searcher_factory(**kwargs)
    481 assert (
    482     kwargs["scheduler"] == "fifo"
    483 ), "This factory needs scheduler = 'fifo' (instead of '{}')".format(
    484     kwargs["scheduler"]
    485 )
    486 # Common objects
--> 487 result = _create_common_objects(**kwargs)
    489 return dict(**result, acquisition_class=EIAcquisitionFunction)

File ~/miniconda3/envs/syne-tune/lib/python3.8/site-packages/syne_tune/optimizer/schedulers/searchers/gp_searcher_factory.py:354, in _create_common_objects(model, is_hypertune, **kwargs)
    350     model = "gp_multitask"
    351 assert (
    352     model == "gp_multitask" or is_hyperband
    353 ), f"model = {model} only together with hyperband_* scheduler"
--> 354 hp_ranges = create_hp_ranges_for_warmstarting(**kwargs)
    355 random_seed, _kwargs = extract_random_seed(kwargs)
    356 # Skip optimization predicate for GP surrogate model

File ~/miniconda3/envs/syne-tune/lib/python3.8/site-packages/syne_tune/optimizer/schedulers/searchers/utils/warmstarting.py:73, in create_hp_ranges_for_warmstarting(**kwargs)
     71     task_param = Categorical(categories=[active_task])
     72     active_config_space = dict(active_config_space, **{task_attr: task_param})
---> 73 return make_hyperparameter_ranges(
     74     config_space, active_config_space=active_config_space, prefix_keys=prefix_keys
     75 )

File ~/miniconda3/envs/syne-tune/lib/python3.8/site-packages/syne_tune/optimizer/schedulers/searchers/utils/hp_ranges_factory.py:33, in make_hyperparameter_ranges(config_space, name_last_pos, value_for_last_pos, active_config_space, prefix_keys)
     26 def make_hyperparameter_ranges(
     27     config_space: Dict,
     28     name_last_pos: Optional[str] = None,
   (...)
     31     prefix_keys: Optional[List[str]] = None,
     32 ) -> HyperparameterRanges:
---> 33     hp_ranges = HyperparameterRangesImpl(
     34         config_space,
     35         name_last_pos=name_last_pos,
     36         value_for_last_pos=value_for_last_pos,
     37         active_config_space=active_config_space,
     38         prefix_keys=prefix_keys,
     39     )
     40     return hp_ranges

File ~/miniconda3/envs/syne-tune/lib/python3.8/site-packages/syne_tune/optimizer/schedulers/searchers/utils/hp_ranges_impl.py:570, in HyperparameterRangesImpl.__init__(self, config_space, name_last_pos, value_for_last_pos, active_config_space, prefix_keys)
    562 def __init__(
    563     self,
    564     config_space: Dict,
   (...)
    568     prefix_keys: Optional[List[str]] = None,
    569 ):
--> 570     super().__init__(
    571         config_space,
    572         name_last_pos,
    573         value_for_last_pos,
    574         active_config_space,
    575         prefix_keys,
    576     )
    577     hp_ranges = []
    578     for name in self.internal_keys:

File ~/miniconda3/envs/syne-tune/lib/python3.8/site-packages/syne_tune/optimizer/schedulers/searchers/utils/hp_ranges.py:82, in HyperparameterRanges.__init__(self, config_space, name_last_pos, value_for_last_pos, active_config_space, prefix_keys)
     80 self.value_for_last_pos = value_for_last_pos
     81 self._set_internal_keys(prefix_keys)
---> 82 self._set_active_config_space(active_config_space)

File ~/miniconda3/envs/syne-tune/lib/python3.8/site-packages/syne_tune/optimizer/schedulers/searchers/utils/hp_ranges.py:106, in HyperparameterRanges._set_active_config_space(self, active_config_space)
    104     self._config_space_for_sampling = self.config_space
    105 else:
--> 106     self._assert_sub_config_space(active_config_space)
    107     self.active_config_space = active_config_space
    108     self._config_space_for_sampling = dict(
    109         self.config_space, **active_config_space
    110     )

File ~/miniconda3/envs/syne-tune/lib/python3.8/site-packages/syne_tune/optimizer/schedulers/searchers/utils/hp_ranges.py:122, in HyperparameterRanges._assert_sub_config_space(self, active_config_space)
    118 same_log_type = is_log_space(v) == is_log_space(
    119     self.config_space[k]
    120 ) and is_reverse_log_space(v) == is_reverse_log_space(self.config_space[k])
    121 same_domain_type = isinstance(v, type(self.config_space[k]))
--> 122 assert (
    123     k in self.config_space
    124     and same_value_type
    125     and same_log_type
    126     and same_domain_type
    127 ), f"active_config_space[{k}] has different type"

AssertionError: active_config_space[problem_maker] has different type
```

**Expected behavior**  
That for both Ordinal and Categorical encoding we should be able to initialise the scheduler.  

**Paste the output of the `pip freeze` command below:**

**Desktop (please complete the following information):**  
 - OS: MacOS Monterey 12.6.1
 - Syne Tune Version v0.3.2

**Screenshots**  

**Additional context**

**Comment by sighellan** [comment]
I upgraded to v0.3.4 and got the same initial assertion error 
(`AssertionError: active_config_space[problem_maker] has different type`).

**Comment by mseeger** [comment]
Hello, I'll have a look at this. It is the case that the "active config space" feature you are trying to use, is not implemented for all Domain types. I see what I can do.

**Comment by mseeger** [comment]
I pushed something into branch `active_space` . Could you test whether this works for you? You’d have to install Syne Tune from source, as explained in `README.md` under “Installing”, then “get the latest version from git”.
Let me know if this works.

BTW: As in your example, you need to use `ordinal(..., kind="equal")`  for this to work. I’ll look into support `kind="nn"` as well, but if this is difficult, I won’t do it now. And if your categories are increasing numbers, then `kind="nn"` is the default.

**Comment by sighellan** [comment]
Thank you very much, this works for me!

**Comment by mseeger** [comment]
Resolved with #503
